src/renderer/animation_utils.rs

```diff
@@ -80,6 +80,71 @@ pub fn ease_point(ease_func: fn(f32) -> f32, start: Point, end: Point, t: f32) -
     }
 }
 
+pub struct CriticallyDampedSpringAnimation {
```

Comment:
This is lovely. Should it be used for the window position animations too?

Reply:
It could, but I think that the cursor animation would benefit more from it. If the cursor animation uses a spring it would much better, follow the row when scrolling. I will try how that works.